pilot/pkg/proxy/envoy/v2/ads.go

```diff
@@ -297,24 +297,15 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 							con.mu.Unlock()
 							continue
 						}
-					} else if discReq.ErrorDetail != nil {
-						// If versions mismatch then we should either have an error detail or no routes if a protocol error has occurred
-						if discReq.ErrorDetail != nil {
-							errCode := codes.Code(discReq.ErrorDetail.Code)
-							adsLog.Warnf("ADS:RDS: ACK ERROR %v %s %s:%s", peerAddr, con.ConID, errCode.String(), discReq.ErrorDetail.GetMessage())
-							incrementXDSRejects(rdsReject, con.node.ID, errCode.String())
-						}
-						continue
```

Comment:
Why remove this ?

Reply:
discReq.ErrorDetail != nil is covered at the beginning.

Reply:
This is a case of version mismatch - I think we should still keep it. ErrorDetail can be present for variety of reasons. Version mismatch is one such case,